src/runtime.js

```diff
@@ -67,6 +68,11 @@ export function adopt(global) {
         name: name,
         implementationClass: implementationClass
       });
+      // Resolve this extension's Service Promise.
+      getService(global, name, () => {
```

Comment:
As for conflicts, the only ones are `amp-analytics-instrumentation` and `amp-dynamic-css-classes`, both of which register their service directly and do not call `AMP.registerElement`.